src/test-jdk17/java/com/fasterxml/jackson/databind/records/RecordWithJsonAnySetterTest.java

```diff
@@ -0,0 +1,110 @@
+        }
+    }
+
+    // Workaround mentioned in [databind#3439]
+    record RecordWorkaroundWithAnySetterMethodWithCanonCtor(int id, Map<String, Object> any) {
+
+        RecordWorkaroundWithAnySetterMethodWithCanonCtor {
+            any = (any != null) ? any : new LinkedHashMap<>();
+        }
+
+        @JsonAnySetter
+        public void addAny(String name, Object value) {
+            any.put(name, value);
+        }
+    }
+
+    // Workaround mentioned in [databind#3439]
+    record RecordWorkaroundWithAnySetterComponentAltCtor(int id, @JsonAnySetter Map<String, Object> any) {
+        @JsonCreator
+        public RecordWorkaroundWithAnySetterComponentAltCtor(@JsonProperty("id") int id) {
+            this(id, new LinkedHashMap<>());
+        }
+    }
+
+    // A proper one without workaround
+    record RecordWithAnySetterComponentCanonCtor(int id, @JsonAnySetter Map<String, Object> any) {
+    }
+
+    private final ObjectMapper MAPPER = newJsonMapper();
+
+    @Test
+    public void testDeserializeWorkaround_WithAnySetterMethod_WithAltConstructor() throws Exception {
+        RecordWorkaroundWithAnySetterMethodWithAltCtor value = MAPPER.readValue(
+                a2q("{'id':123,'any1':'Any 1','any2':'Any 2'}"), RecordWorkaroundWithAnySetterMethodWithAltCtor.class);
+
+        Map<String, Object> expectedAny = new LinkedHashMap<>();
+        expectedAny.put("any1", "Any 1");
+        expectedAny.put("any2", "Any 2");
+        assertEquals(new RecordWorkaroundWithAnySetterMethodWithAltCtor(123, expectedAny), value);
+    }
+
+    @Test
+    public void testDeserializeWorkaround_WithAnySetterMethod_WithCanonicalConstructor() throws Exception {
+        RecordWorkaroundWithAnySetterMethodWithCanonCtor value = MAPPER.readValue(
+                a2q("{'id':123,'any1':'Any 1','any2':'Any 2'}"), RecordWorkaroundWithAnySetterMethodWithCanonCtor.class);
+
+        Map<String, Object> expectedAny = new LinkedHashMap<>();
+        expectedAny.put("any1", "Any 1");
+        expectedAny.put("any2", "Any 2");
+        try {
+            assertEquals(new RecordWorkaroundWithAnySetterMethodWithCanonCtor(123, expectedAny), value);
+            fail("Used to fail, but is now working again");
+        } catch (AssertionFailedError e) {
+            // Used to work, but stopped working starting from 2.18.0 due to [databind#4558]
+            verifyException(e, "[id=123, any={}]");
+        }
+    }
+
+    @Test
+    public void testDeserializeWorkaround_WithAnySetterComponent_WithAltConstructor() throws Exception {
+        try {
+            RecordWorkaroundWithAnySetterComponentAltCtor value = MAPPER.readValue(
+                    a2q("{'id':123,'any1':'Any 1','any2':'Any 2'}"), RecordWorkaroundWithAnySetterComponentAltCtor.class);
+            fail("Used to fail, but is now: " + value);
+        } catch (UnrecognizedPropertyException e) {
+            // Used to work, but stopped working starting from 2.15.0 due to [databind#3737]
```

Comment:
I don't like the idea of tests verifying wrong behavior -- anything that fails should try to test correct behavior but test itself added under "/failing/".

Reply:
But the tests in "/failing/" are for those expected to pass in the future, though.

For these, they're not expected to pass...

Reply:
Ah ok thank you for explanation. This makes sense. 

Whether to remove... if you think fail with exception is the right thing & should be retained, then keeping those makes sense. Otherwise if we don't care, remove.